form/src/main/java/org/dhis2/form/ui/dialog/QRDetailBottomDialog.kt

```diff
@@ -26,34 +42,40 @@ import org.dhis2.form.data.FormFileProvider
 import org.dhis2.form.databinding.QrDetailDialogBinding
 import org.dhis2.form.model.UiRenderType
 import org.hisp.dhis.android.core.arch.helpers.FileResourceDirectoryHelper
+import org.hisp.dhis.mobile.ui.designsystem.component.BottomSheetShell
+import org.hisp.dhis.mobile.ui.designsystem.component.ButtonCarousel
+import org.hisp.dhis.mobile.ui.designsystem.component.CarouselButtonData
+import org.hisp.dhis.mobile.ui.designsystem.component.QrCodeBlock
+import org.hisp.dhis.mobile.ui.designsystem.theme.SurfaceColor
 import timber.log.Timber
 import java.io.File
 import java.io.FileOutputStream
 import java.io.IOException
-import javax.inject.Inject
 
 class
 QRDetailBottomDialog(
     private val value: String,
     private val renderingType: UiRenderType?,
     private val editable: Boolean,
+    useCompose: Boolean,
     private val onClear: () -> Unit,
     private val onScan: () -> Unit,
 ) : BottomSheetDialogFragment() {
+
+    var colorUtils: ColorUtils = ColorUtils()
     companion object {
         const val TAG: String = "QR_DETAIL_DIALOG"
     }
 
-    @Inject
-    lateinit var colorUtils: ColorUtils
-
     private lateinit var binding: QrDetailDialogBinding
     private var qrContentUri: Uri? = null
+    private var useComposeDialog: Boolean = useCompose
```

Comment:
I think there is no need to use another variable, you can directly use `useCompose`

Reply:
I tried that, but it doesn't find the variable from the onCreateView method

Reply:
Maybe because you need `**private val** useCompose: Boolean` in line 60?

Reply:
that was it! thanks I've updated the PR